File: AdventOfCodeCSharp/AdventOfCodeCSharp/InverseCaptcha.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AdventOfCodeCSharp
{
    public static class InverseCaptcha
    {
        public static int CalculateInverseCaptcha(string input)
        {
            var sum = 0;
            for (int i = 0; i < input.Length; i++)
            {
                var currentValueAsInt = int.Parse(input[i].ToString());
                var nextIndex = i == input.Length - 1 ? 0 : i + 1;
                var nextValueAsInt = int.Parse(input[nextIndex].ToString());

                if (currentValueAsInt == nextValueAsInt)
                {
                    sum += currentValueAsInt;
                }
            }

            return sum;
        }

        public static int CalculateInverseCaptchaPartTwo(string input)
        {
            var sum = 0;
            for (int i = 0; i < input.Length; i++)
            {
                var currentValueAsInt = int.Parse(input[i].ToString());
                var desiredIndex = (i + input.Length / 2);
                var nextIndex = desiredIndex > input.Length - 1 ? desiredIndex - input.Length : desiredIndex;
                var nextValueAsInt = int.Parse(input[nextIndex].ToString());

                if (currentValueAsInt == nextValueAsInt)
                {
                    sum += currentValueAsInt;
                }
            }

            return sum;
        }
    }
}

File: AdventOfCodeFSharp/AdventOfCodeFSharp/InverseCaptcha.fs
namespace AdventOfCodeFSharp

module InverseCaptcha = 
    let inline charToInt c = int c - int '0'
    let inline add x y = x + y

    let CalculateInverseCaptcha (input:string) =
        let solution = ref 0
        for i in 0 .. input.Length - 1 do
            let currentValueAsInt = input.[i] |> charToInt
            let nextIndex = if i = input.Length - 1 then 0 else i + 1
            let nextValueAsInt = input.[nextIndex] |> charToInt
            if currentValueAsInt = nextValueAsInt then solution := !solution + currentValueAsInt
        solution.Value

    let CalculateInverseCaptchaPartTwo (input:string) =
        let solution = ref 0
        for i in 0 .. input.Length - 1 do
            let currentValueAsInt = input.[i] |> charToInt
            let desiredIndex = i + input.Length / 2;
            let nextIndex = if desiredIndex > input.Length - 1 then  desiredIndex - input.Length else desiredIndex
            let nextValueAsInt = input.[nextIndex] |> charToInt
            if currentValueAsInt = nextValueAsInt then solution := !solution + currentValueAsInt
        solution.Value